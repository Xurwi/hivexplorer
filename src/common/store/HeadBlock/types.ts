// export interface HeadblockStatde {
//   [page: string]: {
//     scroll: number;
//   };
// }

export interface objAmountPrecisionNai {
  amount: string
  precision: number
  nai: string
}

export interface HeadblockState {

  head_block_number: string,
  head_block_id: string,
  time:string,
  current_witness:string,
  // total_pow:string,
  // num_pow_witnesses:string,
  // virtual_supply:string | objAmountPrecisionNai 
  // current_supply:string | objAmountPrecisionNai 
  // init_hbd_supply:string | objAmountPrecisionNai 
  // current_hbd_supply:string | objAmountPrecisionNai,
  // total_vesting_fund_hive:string | objAmountPrecisionNai,
  // total_vesting_shares:string | objAmountPrecisionNai,
  // total_reward_fund_hive:string | objAmountPrecisionNai,
  // total_reward_shares2:string | objAmountPrecisionNai,
  // pending_rewarded_vesting_shares:string | objAmountPrecisionNai,
  // pending_rewarded_vesting_hive:string | objAmountPrecisionNai, 
  // hbd_interest_rate:string,
  // hbd_print_rate:string,
  // maximum_block_size:string,
  // required_actions_partition_percent:string,
  // current_aslot:string,
  // recent_slots_filled:string,
  // participation_count:string,
  // last_irreversible_block_num:string,
  // vote_power_reserve_rate:string,
  // delegation_return_period:string,
  // reverse_auction_seconds:string,
  // available_account_subsidies:string,
  // hbd_stop_percent:string,
  // hbd_start_percent:string,
  // next_maintenance_time:string,
  // last_budget_time:string,
  // next_daily_maintenance_time:string,
  // content_reward_percent:string,
  // vesting_reward_percent:string,
  // sps_fund_percent:string,
  // sps_interval_ledger:string,
  // downvote_pool_percent:string,
  // proposal_fund_percent: number,
  // dhf_interval_ledger: string | objAmountPrecisionNai,
  // current_remove_threshold:string,
  // early_voting_seconds:string,
  // mid_voting_seconds:string,
  // max_consecutive_recurrent_transfer_failures:string,
  // max_recurrent_transfer_end_date:string,
  // min_recurrent_transfers_recurrence:string,
  // max_open_recurrent_transfers:string
}




export enum ActionTypes {
  SET_HEAD_BLOCK = "@headBlock/SET_DATA"
}

export interface HeadblockAction {
  type: ActionTypes.SET_HEAD_BLOCK;
  data?: HeadblockState;
}

export interface SavePageScrollData {
  pageName: string;
  scrollValue: number;
}

export type Actions = HeadblockAction;