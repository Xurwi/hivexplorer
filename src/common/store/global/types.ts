import { LocationChangeAction } from '../common';

export enum Theme {
  day = 'day',
  night = 'night',
}

export interface Global {
  theme: Theme;
  isMobile: boolean;
  hasKeyChain: boolean;
  lang: string;
}

export enum ActionTypes {
  THEME_CHANGE = '@global/THEME_CHANGE',
  HAS_KEYCHAIN = '@global/HAS_KEYCHAIN',
  LANG_SET = '@global/LANG_SET',
}

export interface HasKeyChainAction {
  type: ActionTypes.HAS_KEYCHAIN;
}

export interface LangSetAction {
  type: ActionTypes.LANG_SET;
  lang: string;
}

export interface ThemeChangeAction {
  type: ActionTypes.THEME_CHANGE;
  theme: Theme;
}

export type Actions = LocationChangeAction | ThemeChangeAction | HasKeyChainAction | LangSetAction;