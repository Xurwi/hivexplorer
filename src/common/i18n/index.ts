// import i18n from 'i18next';

// import moment from 'moment';

// export const langOptions = [
//   {
//     code: 'en-US',
//     name: 'English'
//   },
// ];

// const enUs = require('./locales/en-US.json');
// const resources = {
//   ['en-US']: {
//     translation: enUs
//   },
// };

// i18n.init({
//   resources,
//   fallbackLng: 'en-US',
//   interpolation: {
//     escapeValue: false
//   },
// })

// i18n.on('languageChanged', function (lang) {
//   moment.locale(lang);
// });

// export const _t = (k: string, args = {}) => {
  
//   return i18n.t(k, args);
// };
import LanguageDetector from 'i18next-browser-languagedetector';
import i18n from "i18next";
import { useTranslation, initReactI18next } from "react-i18next";
import HttpApi from 'i18next-http-backend';
i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .use(LanguageDetector)
  .use(HttpApi)
  .init({
    supportedLngs:['en','fr'],
    // the translations
    // (tip move them in a JSON file and import them,
    // or even better, manage them via a UI: https://react.i18next.com/guides/multiple-translation-files#manage-your-translations-with-a-management-gui)
    //lng: document.querySelector('html').lang , // if you're using a language detector, do not define the lng option
    fallbackLng: "en",
    detection:{
      order: [ 'cookie', 'localStorage',  'htmlTag', 'path', 'subdomain'],
      caches:['cookie']
    },
    backend:{
      loadPath: 'assets/locales/{{lng}}/translation.json',
    },
    react:{
      useSuspense:false
    }
  });

export const _t = (k: string, args = {}) => {
  return i18n.t(k, args);
};

  export default i18n;