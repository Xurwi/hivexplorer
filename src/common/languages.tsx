import { frFlag, gbFlag } from "./img/svg";

export const languages = [
  {
    code: "fr",
    name: "Fran√ßais",
    country_code: "fr",
    flagImg: frFlag
  },
  {
    code: "en",
    name: "English",
    country_code: "gb",
    flagImg: gbFlag
  }
];
