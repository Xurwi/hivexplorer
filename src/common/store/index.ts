import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';
import { createBrowserHistory, History } from 'history';

import global from './global';
import persistentPageScroll from './persistent-page-scroll';

export let history: History | undefined;

let reducers = {
  global,
  persistentPageScroll,
};

// create browser history on client side
if (typeof window !== 'undefined') {
  // web
  history = createBrowserHistory();

  // We need a customised history object since history pushes new state for same path.
  // See: https://github.com/ReactTraining/history/issues/470
  // We don't want LOCATION_CHANGE triggered for same path because of structure of out "entries" reducer.

  // get ref of current push function
  const _push = history.push;

  let prevPath: string = history.location.pathname;
  // update previous path once history change
  history.listen(location => {
    prevPath = location.pathname;
  });

  // create a new push function that compares new path and previous path.
  history.push = (pathname: History.Path, state: History.LocationState = {}) => {
    // simple path compare
    if (pathname === prevPath) {
      return;
    }
    _push(pathname.includes('//') ? '/' : pathname, state);
  }

  // scroll to top on every push action
  history.listen((location, action) => {
    if (action === 'PUSH') {
      // Don't scroll to top with anchor links
      if (history!.location.hash !== '') {
        return;
      }

      setTimeout(() => {
        window.scrollTo({
          top: 0
        });
      }, 100);
    }
  });

  // @ts-ignore
  reducers = { router: connectRouter(history), ...reducers };
}

const rootReducer = combineReducers(reducers);

export default rootReducer;

export type AppState = ReturnType<typeof rootReducer>;
