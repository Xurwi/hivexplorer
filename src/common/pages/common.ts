import {AnyAction, bindActionCreators, Dispatch} from "redux";
import {AppState} from "../store";

import {History, Location} from "history";
import {Global} from "../store/global/types";


import {
  toggleTheme,
  setCurrency,
  setLang,
  setLastIndexPath
} from '../store/global';
import {fetchEntries, addEntry, updateEntry, invalidateEntries} from "../store/entries";
import {fetchDiscussion, sortDiscussion, resetDiscussion} from "../store/discussion";
import {addAccount} from "../store/accounts";
import {addCommunity} from "../store/communities";
import {fetchTransactions, resetTransactions} from "../store/transactions";

import {Account, Accounts} from "../store/accounts/types";
import {Community, Communities} from "../store/communities/types";
import {DynamicProps} from "../store/dynamic-props/types";
import {Entries, Entry} from "../store/entries/types";
import {Discussion as DiscussionType, SortOrder} from "../store/discussion/types";
import {Transactions, OperationGroup} from "../store/transactions/types";

export interface PageProps {
    history: History;
    location: Location;

    global: Global;
    toggleTheme: () => void;
    setCurrency: (currency: string, rate: number, symbol: string) => void;
    setLang: (lang: string) => void;
    setLastIndexPath: (path: string | null) => void;

    dynamicProps: DynamicProps;

    entries: Entries;
    fetchEntries: (what: string, tag: string, more: boolean) => void;
    addEntry: (entry: Entry) => void;
    updateEntry: (entry: Entry) => void;
    invalidateEntries: (groupKey: string) => void;

    discussion: DiscussionType;
    fetchDiscussion: (parent_author: string, parent_permlink: string) => void;
    sortDiscussion: (order: SortOrder) => void;
    resetDiscussion: () => void;

    accounts: Accounts;
    addAccount: (data: Account) => void;

    communities: Communities;
    addCommunity: (data: Community) => void;

    transactions: Transactions;
    fetchTransactions: (username: string, group?: OperationGroup | "", start?: number, limit?: number) => void;
    resetTransactions: () => void;

    persistentPageScroll: any;
}

export const pageMapStateToProps = (state: AppState) => ({
    ...state
});

export const pageMapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
    bindActionCreators(
        {
            toggleTheme,
            setCurrency,
            setLang,
            setLastIndexPath,
            fetchEntries,
            addEntry,
            updateEntry,
            invalidateEntries,
            fetchDiscussion,
            sortDiscussion,
            resetDiscussion,
            addAccount,
            addCommunity,
            fetchTransactions,
            resetTransactions,
        },
        dispatch
    );
