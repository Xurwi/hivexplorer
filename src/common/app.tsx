import React, { useEffect } from 'react';
import { Route, Switch } from 'react-router-dom';
import NotFound from './components/404';
import routes from './routes';
import * as ls from './util/local-storage';
import i18n from 'i18next';
import { pageMapDispatchToProps, pageMapStateToProps } from './pages/common';
import { connect } from 'react-redux';
import loadable from '@loadable/component';

const EntryPage = loadable(() => import('./pages/index'));

const App = ({ setLang }: any) => {
  useEffect(() => {
    const currentLang = ls.get('current-language');
    if (currentLang) {
      setLang(currentLang);
      i18n.changeLanguage(currentLang)
    }
  }, []);

  return (
    <>
      {/*<Tracker/>*/}
      <Switch>
        <Route exact={true} path={routes.HOME} component={EntryPage}/>
        <Route component={NotFound}/>
      </Switch>
    </>
  );
};

export default connect(pageMapStateToProps, pageMapDispatchToProps)(App);
