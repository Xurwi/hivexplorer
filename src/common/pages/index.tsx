import { pageMapDispatchToProps, pageMapStateToProps, PageProps } from './common';
import React, { useEffect, useState } from 'react';
import Meta from '../components/meta';
import Theme from '../components/theme';
import { getMetaProps } from '../util/get-meta-props';
import { connect } from 'react-redux';
import { withPersistentScroll } from '../components/with-persistent-scroll';

const Index = (props: PageProps) => {
  const [metaProps, setMetaProps] = useState({});

  useEffect(() => {
    setMetaProps(getMetaProps(props));
  }, []);

  return <>
    <Meta {...metaProps} />
    <Theme global={props.global}/>
    <div>Test page</div>
  </>;
};

export default connect(pageMapStateToProps, pageMapDispatchToProps)(withPersistentScroll(Index));