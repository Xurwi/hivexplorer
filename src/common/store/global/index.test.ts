import reducer, { initialState, themeChangeAct, setLangAct } from "./index";

import { Theme } from "./types";
import { locationChangeAct } from "../common";

let state = initialState;

it("1- default state", () => {
  expect(state).toMatchSnapshot();
});

it("2- location change. should change filter and tag", () => {
  state = reducer(state, locationChangeAct("/trending/bitcoin"));
  expect(state).toMatchSnapshot();
});

it("3- theme change", () => {
  state = reducer(state, themeChangeAct(Theme.night));
  expect(state).toMatchSnapshot();
});

it("4- Set lang act", () => {
  state = reducer(state, setLangAct("tr-TR"));
  expect(state).toMatchSnapshot();
});
