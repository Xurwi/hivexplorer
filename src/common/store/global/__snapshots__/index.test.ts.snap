// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "isMobile": false,
  "lang": "en_US",
  "theme": "day",
}
`;

exports[`2- location change. should change filter and tag 1`] = `
Object {
  "isMobile": false,
  "lang": "en_US",
  "theme": "day",
}
`;

exports[`3- theme change 1`] = `
Object {
  "isMobile": false,
  "lang": "en_US",
  "theme": "night",
}
`;

exports[`4- Set lang act 1`] = `
Object {
  "isMobile": false,
  "lang": "tr-TR",
  "theme": "night",
}
`;
